### ISSUE TITLE
Pillar data not getting populated in state when called from Orchestrate

### ISSUE THREAD

**Issue by adminericjacobs** [saltstack/salt#42695]
When "state.sls teststate" is run directly pillar data is populated, when state is called via Orchestrate pillar data is absent...

```
    Salt Version:
               Salt: 2017.7.0
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: unknown
           dateutil: Not Installed
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.7.2
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 15.3.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.1.4
     
    System Versions:
               dist: centos 7.3.1611 Core
             locale: ANSI_X3.4-1968
            machine: x86_64
            release: 4.9.39-1-MANJARO
             system: Linux
            version: CentOS Linux 7.3.1611 Core
```

Orchestrate state
```
# salt/orch/test.sls
state_run:
  salt.state:
    - tgt: salt-minion1
    - sls:
      - teststate
```

State file
```
# salt/teststate.sls
{% set tomcat_home = salt['pillar.get']('tomcat:home', '') %}

/tmp/teststate.txt:
  file.managed:
    - contents:
      - Tomcat home is {{ tomcat_home }}
```

Orchestrate Run
```
[root@salt-master /]# salt-run state.orch orch.test        
salt-master_master:
----------
          ID: highstate_run
    Function: salt.state
      Result: True
     Comment: States ran successfully. Updating salt-minion1.
     Started: 16:55:34.207992
    Duration: 334.033 ms
     Changes:   
              salt-minion1:
              ----------
                        ID: /tmp/teststate.txt
                  Function: file.managed
                    Result: True
                   Comment: File /tmp/teststate.txt updated
                   Started: 16:55:34.428176
                  Duration: 6.359 ms
                   Changes:   
                            ----------
                            diff:
                                --- 
                                +++ 
                                @@ -1 +1 @@
                                -Tomcat home is /usr/local/tomcat
                                +Tomcat home is
              
              Summary for salt-minion1
              ------------
              Succeeded: 1 (changed=1)
              Failed:    0
              ------------
              Total states run:     1
              Total run time:   6.359 ms

Summary for salt-master_master
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 334.033 ms
```

State run
```
[root@salt-master /]# salt salt-minion1 state.sls teststate
salt-minion1:
----------
          ID: /tmp/teststate.txt
    Function: file.managed
      Result: True
     Comment: File /tmp/teststate.txt updated
     Started: 16:55:55.502320
    Duration: 6.228 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1 +1 @@
                  -Tomcat home is
                  +Tomcat home is /usr/local/tomcat

Summary for salt-minion1
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   6.228 ms
```

**Comment by adminericjacobs** [comment]
I have confirmed that https://github.com/saltstack/salt/pull/42433 fixes this issue

**Comment by adminericjacobs** [comment]
Looks like this fix is included in 2017.7.1 and this bug is a duplicate of https://github.com/saltstack/salt/issues/42403
